scripts/SANS/sans/state/RunDataStateBuilder.py

```diff
@@ -0,0 +1,20 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+
+class RunDataStateBuilder(object):
+    def __init__(self, file_information):
+        self._file_information = file_information
+        # TODO: StateData should be unpicked from the user file and moved into here
```

Comment:
A tiny bit of explanation is required:
StateData was intended way back to hold details we can only know from the actual run file 
However multiple State objects try to do two jobs at the same time by holding data from the user files which are read in. 

This has lead to a lot of kludgy code, since we can't load in a run number (for StateData) until State* objects are built. So SANS uses multiple factories to avoid this and the whole nastiness has spread. 

In a different branch (27664_Add_toml_parser_to_sans) I tried doing this and ended up reverting the individual commits since it is an entire extra PR worth of diffs. This is more a reminder during the maintenance period to sort the whole situation out.

Reply:
Ok that's fine then